src/coreclr/gc/gcpriv.h

```diff
@@ -5105,20 +5105,24 @@ size_t& generation_allocated_since_last_pin (generation* inst)
 #endif //FREE_USAGE_STATS
 
 inline
-float generation_allocator_efficiency (generation* inst)
+uint64_t generation_allocator_efficiency_percent (generation* inst)
 {
-    if ((generation_free_list_allocated (inst) + generation_free_obj_space (inst)) != 0)
-    {
-        return ((float) (generation_free_list_allocated (inst)) / (float)(generation_free_list_allocated (inst) + generation_free_obj_space (inst)));
-    }
-    else
-        return 0;
+    uint64_t free_obj_space = generation_free_obj_space (inst);
+    uint64_t free_list_allocated = generation_free_list_allocated (inst);
+    if (free_list_allocated==0)
+      return 0;
+    return (100 * free_list_allocated) / (free_list_allocated + free_obj_space);
 }
+
 inline
-size_t generation_unusable_fragmentation (generation* inst)
-{
-    return (size_t)(generation_free_obj_space (inst) +
-                    (1.0f-generation_allocator_efficiency(inst))*generation_free_list_space (inst));
+uint64_t generation_unusable_fragmentation (generation* inst)
+{
+    uint64_t free_obj_space = generation_free_obj_space (inst);
+    uint64_t free_list_allocated = generation_free_list_allocated (inst);
+    uint64_t free_list_space = generation_free_list_space (inst);
+    if (free_obj_space==0)
+      return 0;
+    return (free_obj_space + (free_obj_space * free_list_space) / (free_list_allocated + free_obj_space));
```

Comment:
This could use a comment, either that it is based on rewriting of `(100 - efficiency)/100 * free_list_space` or `inefficiency * free_list_space`, being careful in both cases to minimize the impact of the integer division.

Reply:
Thanks for your report. I will make the required change and add the information your want.

Reply:
I would really rather to not have to have the presence of FLT_EPSILON in the GC code at all if I can avoid it :) it's just another thing that someone who's not familiar with FP has to care about. 

but I agree with @markples that these should continue to return size_t, instead of uint64_t. I would also add a comment above `generation_allocator_efficiency_percent` to explain why this is written this way (to accommodate this exception specific to FP which is not a common knowledge AFAIK) instead of the previous way which seems more natural.